openforcefield/tests/test_molecule.py

```diff
@@ -2694,67 +2694,70 @@ def test_chemical_environment_matches_RDKit(self):
-    def test_compute_partial_charges(self):
+    @pytest.mark.parametrize(
+        ("toolkit", "method"),
+        [
+            ("openeye", "mmff94"),
+            ("openeye", "am1bcc"),
+            ("openeye", "am1-mulliken"),
+            ("openeye", "gasteiger"),
+            ("openeye", "am1bccnosymspt"),
+            ("openeye", "am1elf10"),
+            ("openeye", "am1bccelf10"),
+            ("ambertools", "am1bcc"),
+            ("ambertools", "gasteiger"),
+            ("ambertools", "am1-mulliken"),
+        ],
+    )
+    def test_assign_partial_charges(self, toolkit, method):
         """Test computation/retrieval of partial charges"""
         # TODO: Test only one molecule for speed?
         # TODO: Do we need to deepcopy each molecule, or is setUp called separately for each test method?
         import numpy as np
         from simtk import unit
 
         # Do not modify original molecules.
-        molecules = copy.deepcopy(mini_drug_bank())
+        # molecules = copy.deepcopy(mini_drug_bank())
+        # In principle, testing for charge assignment over a wide set of molecules is important, but
+        # I think that's covered in test_toolkits.py. Here, we should only be concerned with testing the
+        # Molecule API, and therefore a single molecule should be sufficient
+        molecule = Molecule.from_smiles("CN1C=NC2=C1C(=O)N(C(=O)N2C)C")
+
+        if toolkit == "openeye":
+            toolkit_registry = ToolkitRegistry(
+                toolkit_precedence=[OpenEyeToolkitWrapper]
+            )
+        elif toolkit == "ambertools":
+            toolkit_registry = ToolkitRegistry(
+                toolkit_precedence=[AmberToolsToolkitWrapper]
+            )
 
-        # Test a single toolkit at a time
-        # Removed  ['amber', 'amberff94'] from OE list, as those won't find the residue types they're expecting
-        toolkit_to_charge_method = {
-            OpenEyeToolkitWrapper: [
-                "mmff",
-                "mmff94",
-                "am1bcc",
-                "am1bccnosymspt",
-                "am1bccelf10",
-            ],
-            AmberToolsToolkitWrapper: ["bcc", "gas", "mul"],
-        }
+        molecule.assign_partial_charges(
+            partial_charge_method=method,
+            toolkit_registry=toolkit_registry,
+        )
+        initial_charges = molecule._partial_charges
+
+        # Make sure everything isn't 0s
+        assert (abs(initial_charges / unit.elementary_charge) > 0.01).any()
+        # Check total charge
+        charges_sum_unitless = initial_charges.sum() / unit.elementary_charge
+        total_charge_unitless = molecule.total_charge / unit.elementary_charge
+        if abs(charges_sum_unitless - total_charge_unitless) > 0.0001:
+            print(
+                "molecule {}    charge_sum {}     molecule.total_charge {}".format(
+                    molecule.name, charges_sum_unitless, total_charge_unitless
+                )
+            )
```

Comment:
(not blocking) Should this `print` be commented out? Or will it only trigger if something's wrong?

Reply:
I'm pretty sure `pytest` only prints prints if a test failed but I'll switch it back to be safe.